coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+                       "use_stderr",
+                       "config_suffix",
+                       "prerequisite_check_command"}
+
+    if not options["use_stdout"] and not options["use_stderr"]:
+        raise ValueError("No output streams provided at all.")
+
+    if options["output_format"] == "corrected":
+        if "diff_severity" in options:
+            if options["diff_severity"] not in RESULT_SEVERITY.reverse:
+                raise TypeError("Invalid value for `diff_severity`: " +
+                                repr(options["diff_severity"]))
+
+        if "diff_message" in options:
+            assert_right_type(options["diff_message"], str, "diff_message")
+
+        allowed_options |= {"diff_severity", "diff_message"}
+    elif options["output_format"] == "regex":
+        if "output_regex" not in options:
+            raise ValueError("No `output_regex` specified.")
+
+        options["output_regex"] = re.compile(options["output_regex"])
+
+        # Don't setup severity_map if one is provided by user or if it's not
+        # used inside the output_regex. If one is manually provided but not
+        # used in the output_regex, throw an exception.
+        if "severity_map" in options:
+            if "severity" not in options["output_regex"].groupindex:
+                raise ValueError("Provided `severity_map` but named group "
+                                 "`severity` is not used in `output_regex`.")
+            assert_right_type(options["severity_map"], dict, "severity_map")
+
+            for key, value in options["severity_map"].items():
+                try:
+                    assert_right_type(key, str, "<severity_map dict-key>")
+                except TypeError:
+                    raise TypeError("The key " + repr(key) + " inside given "
+                                    "severity-map is no string.")
+
+                try:
+                    assert_right_type(value, int, "<severity_map dict-value>")
+                except TypeError:
+                    raise TypeError(
+                        "The value {} for key {} inside given severity-map is "
+                        "no valid severity value.".format(repr(value),
+                                                          repr(key)))
+
+                if value not in RESULT_SEVERITY.reverse:
+                    raise TypeError(
+                        "Invalid severity value {} for key {} inside given "
+                        "severity-map.".format(repr(value), repr(key)))
+
+            # Auto-convert keys to lower-case. This creates automatically a new
+            # dict which prevents runtime-modifications.
+            options["severity_map"] = {
+                key.lower(): value
+                for key, value in options["severity_map"].items()}
+
+        allowed_options |= {"output_regex", "severity_map"}
+    elif options["output_format"] is not None:
+        raise ValueError("Invalid `output_format` specified.")
```

Comment:
Rather than doing this, preferably use the `assert_in` which I'd mentioned above. Makes it cleaner ? (Here I need to go up and check all the values it can take in the `if`s - which is slightly irritating)